lints/lint_rsa_signature_algo_parameter_not_null.go

```diff
@@ -0,0 +1,118 @@
+package lints
+
+/*
+ * ZLint Copyright 2019 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/*******************************************************************************************************
+"RFC5280: RFC 4055, Section 1.2"
+RSA: Encoded algorithm identifier MUST have NULL parameters.
+*******************************************************************************************************/
+
+import (
+	"bytes"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+	"golang.org/x/crypto/cryptobyte"
+	asn1_cryptobyte "golang.org/x/crypto/cryptobyte/asn1"
+)
+
+type rsaEncryptionParamNotNULL struct{}
+
+// byte representation of pkix.Algorithm with OID rsaEncryption and Parameters asn1.NULL, includes BITSTRING tag for public key
+var expectedSPKIAlgoBytes = []byte{0x30, 0xd, 0x6, 0x9, 0x2a, 0x86, 0x48, 0x86, 0xf7, 0xd, 0x1, 0x1, 0x1, 0x5, 0x0, 0x03}
+
+func (l *rsaEncryptionParamNotNULL) Initialize() error {
+	return nil
+}
+
+func (l *rsaEncryptionParamNotNULL) CheckApplies(c *x509.Certificate) bool {
+	// explicitly check for util.OidRSAEncryption, as RSA-PSS or RSA-OAEP certificates might be classified with c.PublicKeyAlgorithm = RSA
+	return c.PublicKeyAlgorithmOID.Equal(util.OidRSAEncryption)
+}
+
+func (l *rsaEncryptionParamNotNULL) Execute(c *x509.Certificate) *LintResult {
+	// Try byte comparison for Algorithm
+	// Determine offset (SEQUENCE + length) first. Notably 1024 encodes length in 2 bytes, 2048 and up in 3 bytes
```

Comment:
I think this comment about how you determine the offset would be better as a comment on the `getOffset` function implementation. It's a detail specific to how that function computes the offset and I think when you're invoking the function it's sufficient to know its a byte offset for comparison.